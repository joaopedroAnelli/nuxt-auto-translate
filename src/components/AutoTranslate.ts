import Vue from 'vue';

export default Vue.extend({
  name: 'AutoTranslate',
  props: {
    tag: {
      type: String,
      default: 'span',
    },
    text: {
      type: String,
      required: true,
    },
  },
  mounted() {
    this.$axios.$post('/api/nuxt-auto-translate/messages', { text: this.text });
  },
  template: `<component :is="tag">{{$t(text)}}</component>`,
});
