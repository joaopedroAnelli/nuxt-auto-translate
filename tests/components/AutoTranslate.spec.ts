import { describe, expect, test, jest } from '@jest/globals';
import { render } from '@testing-library/vue';
import AutoTranslate from '~/components/AutoTranslate';

describe('AutoTranslate', () => {
  const defaultOptions = {
    mocks: {
      $t: (text: string) => text,
      $te: () => true,
      $axios: {
        $post: () => {},
      },
    },
    attrs: {
      text: 'Hello world',
    },
  };

  test('Should simple render', () => {
    const { getByText, container } = render(AutoTranslate, defaultOptions);

    getByText('Hello world');
    expect(container).toMatchSnapshot();
  });

  test('Should change tag', () => {
    const { container } = render(AutoTranslate, defaultOptions);

    expect(container).toMatchSnapshot();
  });

  test('Should fall throug classes', () => {
    const { container } = render(AutoTranslate, defaultOptions);

    expect(container).toMatchSnapshot();
    expect(container.children[0].classList.contains('my-class')).toBeTruthy();
  });

  test('Should send text to server if not exists', () => {
    const sendToServer = jest.fn();
    render(AutoTranslate, {
      ...defaultOptions,
      mocks: {
        ...defaultOptions.mocks,
        $te: () => false,
        $axios: {
          $post: sendToServer,
        },
      },
    });

    expect(sendToServer).toHaveBeenCalled();
    expect(sendToServer).toBeCalledWith('/api/nuxt-auto-translate/messages', {
      text: 'Hello world',
    });
  });

  test('Should not send text to server if exists', () => {
    const sendToServer = jest.fn();
    render(AutoTranslate, {
      ...defaultOptions,
      mocks: {
        ...defaultOptions.mocks,
        $te: () => true,
        $axios: {
          $post: sendToServer,
        },
      },
    });

    expect(sendToServer).not.toBeCalled();
  });
});
