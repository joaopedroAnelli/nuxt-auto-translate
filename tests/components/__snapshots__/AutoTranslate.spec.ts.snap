// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoTranslate Should change tag 1`] = `
<div>
  <p>
    Hello world
  </p>
</div>
`;

exports[`AutoTranslate Should fall throug classes 1`] = `
<div>
  <p
    class="my-class"
  >
    Hello world
  </p>
</div>
`;

exports[`AutoTranslate Should simple render 1`] = `
<div>
  <span>
    Hello world
  </span>
</div>
`;
